### PR TITLE
Fixing issue 1930

### PR DESCRIPTION
**Pull request by dsgouda**
- Updated validation message for ListOperationsNamingWarning
- Split IsXmsPageableOrArrayResponseOperation to get individual type
Fixes https://github.com/Azure/autorest/issues/1930
@veronicagg @vishrutshah PTAL

### CODE REVIEW

File: src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs [hunk 1]
```diff
@@ -36,11 +36,8 @@ public static bool IsTrackedResource(Schema schema, Dictionary<string, Schema> d
         }
 
         // determine if the operation is xms pageable or returns an object of array type
-        public static bool IsXmsPageableOrArrayResponseOperation(Operation op, ServiceDefinition entity)
+        public static bool IsArrayTypeResponseOperation(Operation op, ServiceDefinition entity)
```
Comment (veronicagg) [inline src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs]: @sarangan12 just did a very similar thing, once he merges, you should share this. https://github.com/Azure/autorest/pull/2002/files

File: src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs [hunk 2]
```diff
@@ -70,6 +67,8 @@ public static bool IsXmsPageableOrArrayResponseOperation(Operation op, ServiceDe
             return false;
         }
 
+        public static bool IsXmsPageableResponseOperation(Operation op) => (op.Extensions.GetValue<object>(XmsPageable) != null);
```
Comment (veronicagg) [inline src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs]: same as above comment, please take a look at https://github.com/Azure/autorest/pull/2002/files

**Comment by sarangan12** [comment]
@dsgouda I have already splitted the xmspageableorarrayresponse operation for a different task. Please merge your changes with origin/master.

**Comment by dsgouda** [comment]
@veronicagg I have refactored some of Sarangan's changes, please take a look

File: src/core/AutoRest.Core/Properties/Resources.resx [hunk 3]
```diff
@@ -384,4 +384,7 @@ Modelers:
   <data name="ArmPropertiesBagValidationMessage" xml:space="preserve">
     <value>Top level property names should not be repeated inside the properties bag for ARM resource "{0}". Properties [{1}] conflict with ARM top level properties. Please rename these.</value>
   </data>
+  <data name="ListOperationsNamingWarningMessage" xml:space="preserve">
+    <value>Since operation {0} response is {1} type, it should be named as "list_*"</value>
```
Comment (veronicagg) [inline src/core/AutoRest.Core/Properties/Resources.resx]: is "type" mentioned here an actual type or a model definition? If it's a model definition, please update the message to reflect that.

**Pull request merged**